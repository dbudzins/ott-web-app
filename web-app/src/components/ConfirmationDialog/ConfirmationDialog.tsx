import React from 'react';
import { useTranslation } from 'react-i18next';

import Dialog from '../Dialog/Dialog';
import Button from '../Button/Button';

import styles from './ConfirmationDialog.module.scss';

type Props = {
  open: boolean;
  title: string;
  body: string;
  onConfirm: () => void;
  onClose: () => void;
};

const ConfirmationDialog: React.FC<Props> = ({ open, title, body, onConfirm, onClose }: Props) => {
  const { t } = useTranslation('common');

  return (
    <Dialog open={open} onClose={onClose}>
      <h2 className={styles.title}>{title}</h2>
      <p className={styles.body}>{body}</p>
      <Button className={styles.confirmButton} label={t('confirmation_dialog.confirm')} variant="contained" color="primary" onClick={onConfirm} fullWidth />
      <Button label={t('confirmation_dialog.close')} variant="outlined" onClick={onClose} fullWidth />
    </Dialog>
  );
};

export default ConfirmationDialog;
