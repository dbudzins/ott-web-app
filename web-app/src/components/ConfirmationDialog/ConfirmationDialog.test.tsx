import React from 'react';
import { render } from '@testing-library/react';

import ConfirmationDialog from './ConfirmationDialog';

vi.mock('../Dialog/Dialog', () => ({
  default: 'div',
}));

describe('<ConfirmationDialog>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ConfirmationDialog body="Body" title="Title" open={true} onConfirm={vi.fn()} onClose={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
