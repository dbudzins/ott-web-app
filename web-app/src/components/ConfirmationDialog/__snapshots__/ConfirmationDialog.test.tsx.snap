// Vitest Snapshot v1

exports[`<ConfirmationDialog> > renders and matches snapshot 1`] = `
<div>
  <div
    open=""
  >
    <h2
      class="title"
    >
      Title
    </h2>
    <p
      class="body"
    >
      Body
    </p>
    <button
      class="button confirmButton primary fullWidth"
    >
      <span>
        confirmation_dialog.confirm
      </span>
    </button>
    <button
      class="button default outlined fullWidth"
    >
      <span>
        confirmation_dialog.close
      </span>
    </button>
  </div>
</div>
`;
